edgesite/pkg/controller/controller/upstream.go

```diff
@@ -0,0 +1,589 @@
+					} else {
+						log.LOGGER.Infof("message: %s, update pod status successfully, namespace: %s, name: %s", msg.GetID(), updatedPod.Namespace, updatedPod.Name)
+						if updatedPod.DeletionTimestamp != nil && (status.Phase == v1.PodSucceeded || status.Phase == v1.PodFailed) {
+							if uc.isPodNotRunning(status.ContainerStatuses) {
+								if err := uc.kubeClient.CoreV1().Pods(updatedPod.Namespace).Delete(updatedPod.Name, metaV1.NewDeleteOptions(0)); err != nil {
+									log.LOGGER.Warnf("message: %s, graceful delete pod failed with error: %s, namespace: %s, name: %s", msg.GetID, err, updatedPod.Namespace, updatedPod.Name)
+								}
+								log.LOGGER.Infof("message: %s, pod delete successfully, namespace: %s, name: %s", msg.GetID(), updatedPod.Namespace, updatedPod.Name)
+							}
+						}
+					}
+				}
+
+			default:
+				log.LOGGER.Infof("pod status operation: %s unsupported", msg.GetOperation())
+			}
+			log.LOGGER.Infof("message: %s process successfully", msg.GetID())
+		case <-stop:
+			log.LOGGER.Infof("stop updatePodStatus")
+			running = false
+		}
+	}
+}
+
+func (uc *UpstreamController) updateNodeStatus(stop chan struct{}) {
+	running := true
+	for running {
+		select {
+		case msg := <-uc.nodeStatusChan:
+			log.LOGGER.Infof("message: %s, operation is: %s, and resource is %s", msg.GetID(), msg.GetOperation(), msg.GetResource())
+			nodeStatusRequest := &edgeapi.NodeStatusRequest{}
+			data, err := json.Marshal(msg.GetContent())
+
+			if err != nil {
+				log.LOGGER.Warnf("message: %s process failure, marshal message content with error: %s", msg.GetID(), err)
+				continue
+			}
+			err = json.Unmarshal(data, nodeStatusRequest)
+			if err != nil {
+				log.LOGGER.Warnf("message: %s process failure, unmarshal marshaled message content with error: %s", msg.GetID(), err)
+				continue
+			}
+
+			namespace, err := messagelayer.GetNamespace(msg)
+			if err != nil {
+				log.LOGGER.Warnf("message: %s process failure, get namespace failed with error: %s", msg.GetID(), err)
+				continue
+			}
+			name, err := messagelayer.GetResourceName(msg)
+			if err != nil {
+				log.LOGGER.Warnf("message: %s process failure, get resource name failed with error: %s", msg.GetID(), err)
+				continue
+			}
+
+			switch msg.GetOperation() {
+			case model.UpdateOperation:
+				getNode, err := uc.kubeClient.CoreV1().Nodes().Get(name, metaV1.GetOptions{})
+				if errors.IsNotFound(err) {
+					log.LOGGER.Warnf("message: %s process failure, node %s not found", msg.GetID(), name)
+					continue
+				}
+
+				if err != nil {
+					log.LOGGER.Warnf("message: %s process failure with error: %s, name: %s", msg.GetID(), err, namespace, name)
+					continue
+				}
+
+				// In case the status stored at metadata service is outdated, update the heartbeat automatically
+				/*for i := range nodeStatusRequest.Status.Conditions {
```

Comment:
Why it is period commented?
If it's intended, please describe when to uncomment this.